packages/devtools/cli/src/commands/agent/run.ts

```diff
@@ -52,7 +52,7 @@ export default class Run extends BaseCommand<typeof Run> {
       await agent.monitorEpochs({ limit });
     }
 
-    // NOTE: This is currently called by the agent's forever daemon.
+    // NOTE: This is not currently called by the agent's forever daemon.
```

Comment:
`forever.startDaemon` literally spawns a Node process that invokes the `dx agent run` handler (i.e., this method). 

Therefore `agent.start` is ALWAYS invoked via this handler. Sometimes by the user calling this directly and sometimes via the forever daemon. BUT only the daemon detects the lock. So running the agent manually and running it at the same time via the daemon will be in conflict.

So my expectation was that the locking should be at the `agent` not `daemon` level?

Also... the daemon doesn't know about agents that are running manually via `dx agent run`. Perhaps we should ONLY consider running agents via the daemon and have a `--foreground` flag that blocks the CLI? @dmaretskyi ?

Reply:
Agent level for sure, I want `daemon` to fail if you already running the agent yourself